File: go/wam/1616.go
const WamChatFilterEvent = 1616 // channel: regular

type ChatFilterEventEvent struct {
	ActionType CHATFILTERACTIONTYPES `wam:"action_type"` // field 1
	FilterType CHATFILTERTYPES `wam:"filter_type"` // field 2
	SessionId int64 `wam:"session_id"` // field 3
	SearchResultType CHATSEARCHRESULTTYPE `wam:"search_result_type"` // field 4
	TargetScreen CHATFILTERTARGETSCREEN `wam:"target_screen"` // field 5
	ActivitySessionId string `wam:"activity_session_id"` // field 6
	Metadata string `wam:"metadata"` // field 7
	SearchQueryId string `wam:"search_query_id"` // field 8
	SearchRequestId string `wam:"search_request_id"` // field 9
	PredefinedId int64 `wam:"predefined_id"` // field 10
	LabelName string `wam:"label_name"` // field 11
	ListId int64 `wam:"list_id"` // field 12
	ListIndex int64 `wam:"list_index"` // field 13
	ListType LISTTYPE `wam:"list_type"` // field 14
	ThreadId string `wam:"thread_id"` // field 15
}

File: rust/wam/1616.rs
pub const WAM_CHAT_FILTER_EVENT: u32 = 1616;

#[derive(Debug, Default)]
pub struct ChatFilterEventEvent {
    /// field 1, wire `action_type`
    pub action_type: Option<CHATFILTERACTIONTYPES>,
    /// field 2, wire `filter_type`
    pub filter_type: Option<CHATFILTERTYPES>,
    /// field 3, wire `session_id`
    pub session_id: Option<i64>,
    /// field 4, wire `search_result_type`
    pub search_result_type: Option<CHATSEARCHRESULTTYPE>,
    /// field 5, wire `target_screen`
    pub target_screen: Option<CHATFILTERTARGETSCREEN>,
    /// field 6, wire `activity_session_id`
    pub activity_session_id: Option<String>,
    /// field 7, wire `metadata`
    pub metadata: Option<String>,
    /// field 8, wire `search_query_id`
    pub search_query_id: Option<String>,
    /// field 9, wire `search_request_id`
    pub search_request_id: Option<String>,
    /// field 10, wire `predefined_id`
    pub predefined_id: Option<i64>,
    /// field 11, wire `label_name`
    pub label_name: Option<String>,
    /// field 12, wire `list_id`
    pub list_id: Option<i64>,
    /// field 13, wire `list_index`
    pub list_index: Option<i64>,
    /// field 14, wire `list_type`
    pub list_type: Option<LISTTYPE>,
    /// field 15, wire `thread_id`
    pub thread_id: Option<String>,
}